auth/tests/api/users.rs

```diff
@@ -169,6 +169,80 @@ async fn successful_upgrade_to_pro() {
     );
 }
 
+#[tokio::test]
+async fn downgrade_from_cancelledpro() {
+    let app = app().await;
+
+    // Wait for the mocked Stripe server to start.
+    tokio::task::spawn(app.mocked_stripe_server.clone().serve());
+    tokio::time::sleep(Duration::from_secs(1)).await;
+
+    // Create user with basic tier
+    let response = app.post_user("test-user", "basic").await;
+    assert_eq!(response.status(), StatusCode::OK);
+
+    // Upgrade user to pro
+    let response = app
+        .put_user("test-user", "pro", MOCKED_CHECKOUT_SESSIONS[3])
+        .await;
+    assert_eq!(response.status(), StatusCode::OK);
+
+    // Cancel subscription
+    let response = app.put_user("test-user", "cancelledpro", "").await;
+    assert_eq!(response.status(), StatusCode::OK);
+
+    // Trigger status change to canceled
+    let response = app.get_user("test-user").await;
+    assert_eq!(response.status(), StatusCode::OK);
```

Comment:
Can we check right after that the tier is still `cancelledpro`?

Reply:
Ah, it looks like you're testing this in a dedicated test case. Just wondering, is it correct to assume for this test that the first `get_user` call will return the tier as `cancelledpro`? If yes, than I think it is worth checking (even if you check specifically in another test for it).